### ISSUE TITLE
Aura cannot build dropbox AUR package

### ISSUE THREAD

**Issue by apirogov** [fosskers/aura#955]
When I do it manually via makepkg, it works fine. With `aura -A dropbox` I get:

```
...
==> Beginne prepare()...
[dropbox]                         Generating desktop file... no
dropbox.desktop already exists. Use -f as the first argument to overwrite it.
==> FEHLER: Ein Fehler geschah in prepare().
    Breche ab...
aura :: Package failed to build, citing:

  makepkg failed.

aura :: Action cancelled.
```

**Comment by fosskers** [comment]
Interesting, this seems to occur during the second build attempt. I will investigate more.

**Comment by chris-a-mw** [comment]
Not sure if it helps but I had to add a '-f' to the PKGBUILD prepare step when installing from Arch AUR:
`prepare() {
  gendesk -f --pkgname="$pkgname" --pkgdesc="$pkgdesc" --categories=Network PKGBUILD
}`

After running with that flag once, I was able to remove the flag and run additional upgrade installs without the '-f' flag.

**Comment by fosskers** [comment]
Perhaps then this is something we could suggest to the upstream packager.

**Comment by apirogov** [comment]
But what is the difference between how aura builds vs. running makepkg manually? I would expect that aura should be able to build whatever I can build with makepkg without any additional "patches".

**Comment by fosskers** [comment]
Exactly, there shouldn't be a difference. Aura is just invoking `makepkg` as-is. The issue is that when the build directory is not "clean", some `configure` settings seem to be conflicting.

**Comment by ultimatespirit** [comment]
This issue had been annoying me for a while but I just kept `--hotedit`-ing it away due to lack of time. I've gone ahead and actually requested the AUR maintainer of `dropbox` fix this: https://aur.archlinux.org/packages/dropbox#comment-1050464

That being said, I think this could be an opportunity for some error message improvements. We now know that without `--clean` specified to `aura -A` certain PKGBUILDs have the potential to fail if they're non-hermitic. How about instead of the error message we currently have:
```
==> ERROR: A failure occurred in prepare().
    Aborting...
aura :: Package failed to build, citing:

  makepkg failed.

aura :: Action cancelled.
```

We expand cases where it failed to mention `--clean` and filing a comment against the relevant AUR if that fixes it (or perhaps prompt asking if the user would like to try again with `--clean`?).

I.e. something like:
```
==> ERROR: A failure occurred in prepare().
    Aborting...
aura :: Package failed to build, citing:

  makepkg failed.

aura :: Action cancelled.
aura :: Consider rerunning with `--clean` set, 
           if that succeeds the AUR package has an issue with building
           in a non-clean directory and should be fixed. Consider contacting the maintainer.
```

...which normally would suffice however, because the build cannot succeed in this state the `--clean` command _does not work_, as it only triggers upon success. As far as I can tell there is no method from `aura` alone to salvage this situation that isn't `--hotedit` (requiring a user have figured out what specific thing to fix in the pkgbuild to avoid non-hermicitity).

Accordingly, I'll file a new issue about ways to clean `cache/aura/builds` without requiring a successful build.
